libpod/util.go

```diff
@@ -29,6 +33,87 @@ const (
 	unknownPackage = "Unknown"
 )
 
+// Handle volume options from CLI.
+// Parse "o" option to find UID, GID, Size.
+func VolumeOptions(opts map[string]string) ([]VolumeCreateOption, error) {
+	libpodOptions := []VolumeCreateOption{}
+	volumeOptions := make(map[string]string)
+
+	for key, value := range opts {
+		switch key {
+		case "o":
+			// o has special handling to parse out UID, GID.
+			// These are separate Libpod options.
+			splitVal := strings.Split(value, ",")
+			finalVal := []string{}
+			for _, o := range splitVal {
+				// Options will be formatted as either "opt" or
+				// "opt=value"
+				splitO := strings.SplitN(o, "=", 2)
+				switch strings.ToLower(splitO[0]) {
+				case "size":
+					size, err := units.FromHumanSize(splitO[1])
+					if err != nil {
+						return nil, errors.Wrapf(err, "cannot convert size %s to integer", splitO[1])
+					}
+					libpodOptions = append(libpodOptions, WithVolumeSize(uint64(size)))
+					finalVal = append(finalVal, o)
+					// set option "SIZE": "$size"
+					volumeOptions["SIZE"] = splitO[1]
+				case "inodes":
+					inodes, err := strconv.ParseUint(splitO[1], 10, 64)
```

Comment:
This blows up if split0[1] does not exist.

Reply:
This is just a move of exact function from pkg/domain/infra/abi/parse/parse.go but fixed.